### DISCUSSION TITLE
Feature Request: hui-map-card allow filtering history by gps_accuracy

### DISCUSSION THREAD

**Discussion by legomind** [home-assistant/frontend#7601] [Other feature requests]
At certain locations, the gps signal is terrible, so when home assistant app reports my location, it has a poor accuracy (200m+)

This causes the map to be littered with points & lines making it hard to read.
When plotting the history of the entity, can we allow filtering of each point to require a minimum accuracy or perhaps ignore points that are not a certain distance from the previous point?

**Comment by KTibow** [comment]
Any ideas on how hard this would be to implement? If all that this needs is wrapping a block of code in an `if`, then I can do that.

**Reply by legomind** [reply]
I figured a conditional could be added to the LatLong aggregate

**Reply by legomind** [reply]
Or perhaps a filter on the history array

**Reply by KTibow** [reply]
Looks like I need to change this block:
https://github.com/home-assistant/frontend/blob/cc1d50491b42f1ba86b62898db95f872d62aa0e5/src/panels/lovelace/cards/hui-map-card.ts#L434-L444
The harder part is handling the threshold: If it doesn't have a `gps_accuracy` at all, should it still be displayed? Should a configuration variable be added? In that case, does the UI editor need to be updated?

**Reply by legomind** [reply]
I personally don't care if the UI editor is updated. On the matter of not having gps_accuracy, I would have a seperate option to include those points anyway, defaulting to true.

**Reply by KTibow** [reply]
What is `gps_accuracy` again? What would be a good default threshold? Is lower better accuracy or worse?

**Reply by legomind** [reply]
gps_accuracy is the estimated meters to which the coordinates are accurate. The smaller number the better. 5 is generally considered the best you will get. In buildings the best you will get is 50.

**Reply by KTibow** [reply]
🚀 #7618

**Reply by legomind** [reply]
Wow you did not waste any time!

**Reply by KTibow** [reply]
It's a draft, still figuring it out.
